### PR TITLE
Add validation for tags' names

### PR DESCRIPTION
**Pull request by lubosmj**
closes #7506

### CODE REVIEW

File: pulp_container/app/serializers.py [hunk 1]
```diff
@@ -293,6 +296,15 @@ def validate(self, data):
         """
         new_data = super().validate(data)
 
+        if not re.match(VALID_TAG_REGEX, data["tag"]):
+            raise serializers.ValidationError(
+                _(
+                    "The provided tag '{}' is not valid. A tag may contain lowercase and uppercase "
+                    "ASCII alphabetic characters, digits, underscores, periods, and dashes. A tag "
+                    "must not start with a period or a dash.".format(data["tag"])
+                )
```
Comment (mdellweg) [inline pulp_container/app/serializers.py:305]:
```suggestion
                _(
                    "The provided tag '{}' is not valid. A tag may contain lowercase and uppercase "
                    "ASCII alphabetic characters, digits, underscores, periods, and dashes. A tag "
                    "must not start with a period or a dash."
                ).format(data["tag"])
```

Comment (mdellweg) [reply pulp_container/app/serializers.py:305]: Do you think, https://www.django-rest-framework.org/api-guide/fields/#regexfield helps here?

File: pulp_container/app/serializers.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 from gettext import gettext as _
 import os
+import re
```
Comment (mdellweg) [inline pulp_container/app/serializers.py:3]: Seems unneeded now.

**Review by mdellweg — APPROVED** [review]
One comment pending.

**Review by mdellweg — APPROVED** [review]

**Review by mdellweg — COMMENTED** [review]
I thought i saw a unit test for this in the last version. Where has it gone?

File: pulp_container/app/serializers.py [hunk 3]
```diff
@@ -253,7 +255,18 @@ class TagOperationSerializer(serializers.Serializer):
     A base serializer for tagging and untagging manifests.
     """
 
-    tag = serializers.CharField(required=True, help_text="A tag name")
+    tag = serializers.RegexField(
+        regex=VALID_TAG_REGEX,
```
Comment (ipanova) [inline pulp_container/app/serializers.py:259]: :+1:

**Comment by ipanova** [comment]
> I thought i saw a unit test for this in the last version. Where has it gone?

Not sure about unit test, but we have some functional tests around tagging, we can extend those https://github.com/pulp/pulp_container/blob/master/pulp_container/tests/functional/api/test_tagging_images.py

**Comment by mdellweg** [comment]
I'd prefer unittests, serializers are really easy to test that way, and they need a lot less resources to run.

For inspiration:
https://github.com/pulp/pulp_container/blob/master/pulp_container/tests/unit/test_serializers.py#L7

**Comment by ipanova** [comment]
Yep, agreed on those statements

File: pulp_container/tests/unit/test_serializers.py [hunk 4]
```diff
@@ -63,3 +63,27 @@ def test_invalid_push_version_data(self):
...
+        self.assertTrue(serializer.is_valid(), serializer.errors)
+
+    def test_invalid_tag(self):
+        """Test the serializer while passing an invalid tag."""
+        serializer = TagOperationSerializer(
+            data={"tag": ".invalid-tag", "repository": self.repository}
+        )
+        self.assertFalse(serializer.is_valid())
+        self.assertIn("tag", serializer.errors)
+        self.assertIn("tag is not valid", str(serializer.errors["tag"][0]))
```
Comment (mdellweg) [inline pulp_container/tests/unit/test_serializers.py:89]: You nailed it!

**Review by mdellweg — APPROVED** [review]

**Review by ipanova — APPROVED** [review]

**Pull request merged**